Products/GenericSetup/tool.py

```diff
@@ -939,16 +942,36 @@ def getDependenciesForProfile(self, profile_id):
 
         dependencies = profile_info.get('dependencies', ())
 
-        for dependency_id in dependencies:
+        if not ignore_broken:
 
-            if not self.profileExists(dependency_id):
+            for dependency_id in dependencies:
+
+                if not self.profileExists(dependency_id):
 
-                raise KeyError('Profile "%s" requires the \
-                    dependency-profile "%s", which does not exist.'
-                    % (profile_id, dependency_id))
+                    raise KeyError('Profile "%s" requires the \
+                        dependency-profile "%s", which does not exist.'
+                        % (profile_id, dependency_id))
 
         return dependencies
 
+    @security.protected(ManagePortal)
+    def getBrokenDependencies(self, profile_id):
+        """Return referenced dependency-ids, which do not exist."""
+        broken_dependencies = []
+        dependency_ids = self.getDependenciesForProfile(profile_id,
+                                                ignore_broken=True)
+        for dependency_id in dependency_ids:
+            if not self.profileExists(dependency_id):
+                broken_dependencies.append(dependency_id)
+        return broken_dependencies
+
+    @security.protected(ManagePortal)
+    def hasDependencies(self, profile_id):
+        return len(self.getDependenciesForProfile(profile_id)) > 0
```

Comment:
I see you've added a test for `hasBrokenDependencies` (and therefore cover `getBrokenDependencies`, but does `hasDependencies` need to exist?  If so, it needs tests (empty deps, normal deps, one or more broken deps).

Reply:
Oh gee, that's a leftover not used anywhere, I remove it.